CMakeLists.txt

```diff
@@ -236,7 +230,12 @@ if(MIOPEN_HIP_COMPILER MATCHES ".*clang\\+\\+$")
         PATH_SUFFIXES bin
         PATHS /opt/rocm/llvm
     )
+    # we are using hip-clang and need to remove repeated hip compiler args
+    string(REGEX REPLACE "-mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false" "" HIP_COMPILER_FLAGS "${HIP_COMPILER_FLAGS}")
```

Comment:
Why is this is regex? This seems very specific to how the flags maybe added, which could be wrong if the way the flags are added change.

Wouldn't it be better to find the last `-mllvm -amdgpu-early-inline-all=\w+`, then remove all `-mllvm -amdgpu-early-inline-all=\w+` and then add the last item found back?

Reply:
@JehandadKhan Resolved?

Reply:
The issue is not that different make files add them repeatedly, the issue is that the `hip-clang` adds these flags by default! Thus even if they exist only once here, the repetition would still happen. 

I agree with your point of these arguments being too specific and I will update the regex to handle that. 

Please note that we are only working around hip-config.cmake since they should not be added by that file in the first place and we would not have to do this at all.

Reply:
I don't follow the discussion here. Is that hip-clang adding the same flag twice, so you are removing the duplicated flags? If that is the case, I think it should be fixed by compiler team.

Reply:
@JehandadKhan 
> I don't follow the discussion here. Is that hip-clang adding the same flag twice, so you are removing the duplicated flags? If that is the case, I think it should be fixed by compiler team.